### PR TITLE
BXMSDOC-7577 QE Review: School timetable on Quarkus and other OptaPlanner stories

### PR DESCRIPTION
**Pull request by emmurphy1**
[RHPAM](http://file.ork.redhat.com/~emmurphy/BXMSDOC-7577-pam/#assembly-optaplanner-school-timetable-quarkus_optaplanner-quickstarts)

[RHDM](http://file.ork.redhat.com/~emmurphy/BXMSDOC-7577-dm/#assembly-optaplanner-school-timetable-quarkus_optaplanner-quickstarts)

### CODE REVIEW

**Comment by dupliaka** [comment]
8.1. CREATING THE SCHOOL TIMETABLE PROJECT
The command should be :
mvn io.quarkus:quarkus-maven-plugin:1.11.6.Final-redhat-00001:create \
    -DprojectGroupId=com.example \
    -DprojectArtifactId=optaplanner-quickstart  \
    -Dextensions="resteasy,resteasy-jackson,optaplanner-quarkus,optaplanner-quarkus-jackson" \
    -DplatformGroupId=com.redhat.quarkus \
    -DplatformVersion=1.11.6.Final-redhat-00001 \
    -DnoExamples
(The platform version for Quarkus is 1.11.6.Final-redhat-00001 not 1.11.6.Final.redhat-00001)

**Comment by dupliaka** [comment]
Table 8.1. Properties used in the mvn io.quarkus:quarkus-maven-plugin:1.11.6.Final.redhat-00001:create command
Please change the command mvn quarkus:list-extension to `mvn quarkus:list-extensions`

**Comment by dupliaka** [comment]
8.1. CREATING THE SCHOOL TIMETABLE PROJECT
<quarkus-plugin.version>1.11.6.Final.redhat-00001</quarkus-plugin.version>
should be changed to <quarkus-plugin.version>1.11.6.Final-redhat-00001</quarkus-plugin.version>
...
	<quarkus.platform.version>1.11.6.Final.redhat-00001.Final-redhat-00001</quarkus.platform.version> to <quarkus.platform.version>1.11.6.Final-redhat-00001</quarkus.platform.version>

**Comment by dupliaka** [comment]
There are still no TimeTableResource class that allows to use rest endpoint "solve" in the demonstration example. 
see the section https://quarkus.io/guides/optaplanner#create-the-solver-service

**Comment by emmurphy1** [comment]
@duplia, added.

**Comment by dupliaka** [comment]
School timetable on Quarkus 
Thanks for the changes 
 LGTM

**Review by dupliaka — APPROVED** [review]

**Comment by dupliaka** [comment]
Chapter 10. Red Hat build of OptaPlanner on Spring Boot: a school time table quick start guide
Prerequisites
We need to keep it common for both quickstarts, Quarkus and Spring:
 OpenJDK 11 or later is installed. Red Hat build of Open JDK is available from the Software Downloads page in the Red Hat Customer Portal (login required).
Apache Maven 3.6 or higher is installed. Maven is available from the Apache Maven Project website. 
An IDE, such as IntelliJ IDEA, VSCode, Eclipse, or NetBeans is available.

**Comment by dupliaka** [comment]
4.4. Dinner party should be removed

**Comment by dupliaka** [comment]
1. We need to add "Defining constraints for employee rostering using constraint streams" to "6.2.4. Employee rostering constraints"

After we moved implementation to Quarkus - we start to use constraint streams by default in VRP and ERP - please contact Jiří Locker for further details


2. I suggest to change last sentence to 11.1.4. Score Configuration to
"You can define constraints using the plain Java or Drools or OptaPlanner’s ConstraintStream API. " The section contains explanation on defining constraints in Optaplanner so we need to add constraint streams there as well. I propouse by now to mention it in the sentence and give a link with reference to "How to use constraint streams" like here https://access.redhat.com/documentation/en-us/red_hat_process_automation_manager/7.8/html/using_red_hat_business_optimizer_with_spring_boot/business-optimizr-define-contraints-proc_business-optimizer-spring-boot

**Comment by dupliaka** [comment]
13.2.1. Deploying the application using the provided script
Section procedure
Change step 3 to
6. Navigate to the optaweb-8.5.0.Final-redhat-00001/optaweb-employee-rostering folder.

**Comment by dupliaka** [comment]
13.2.1. Deploying the application using the provided script
Need to detail what "Maven and a Java Development Kit are installed on your local system. "
we can reuse quick start sections:
   OpenJDK 11 or later is installed. Red Hat build of Open JDK is available from the Software Downloads page in the Red Hat Customer Portal (login required).
    Apache Maven 3.6 or higher is installed. Maven is available from the Apache Maven Project website.

**Comment by dupliaka** [comment]
13.2.1. Deploying the application using the provided script
p 10 Change `./provision.sh` to `./runOnOpenShift.sh`

**Comment by dupliaka** [comment]
Table 2. Upgrade impact labels
Remove Impl detail, recommended, readme row

**Comment by dupliaka** [comment]
10.1. Downloading and building the Spring Boot school timetable quick start
12.2.1. Preparing deployment files
14.2. Download and build the OptaWeb Vehicle Routing deployment files
10.1. Downloading and building the Spring Boot school timetable quick start
This sections has common pre requirement - setup maven repo with following steps:

```
Procedure

    1. Navigate to the Software Downloads page in the Red Hat Customer Portal (login required), and select the product and version from the drop-down options:
        Product: Red Hat Process Automation Manager
        Version: 7.11 
    2. Download Red Hat Process Automation Manager 7.11 Maven Repository (kogito-1.5.0-maven-repository.zip).
    3. Extract the kogito-1.5.0-maven-repository.zip file.
    4. Copy the contents of the kogito-1.5.0.GA-maven-repository/maven-repository subdirectory into the ~/.m2/repository directory. 
```  
Could we move it in separate section and refer to it?

**Comment by dupliaka** [comment]
14.3.1. Run the OptaWeb Vehicle Routing runLocally.sh script in quick start mode
Change  optaweb-vehicle-routing-distribution-7.52.0.Final-redhat-00004/sources to 
kogito-1.5.0-decision-services-quickstarts/optaweb-8.5.0.Final-redhat-00001/optaweb-vehicle-routing
The same for
14.3.2. Run the OptaWeb Vehicle Routing runLocally.sh script in interactive mode 
14.3.3. Run the OptaWeb Vehicle Routing runLocally.sh script in non-interactive mode
14.5. Run OptaWeb Vehicle Routing on Red Hat OpenShift Container Platform
14.5.1. Updating the deployed OptaWeb Vehicle Routing application with local changes
14.6.3. Creating custom data sets with OptaWeb Vehicle Routing
14.7.2.1. Running the OptaWeb Vehicle Routing back-end module

**Comment by dupliaka** [comment]
14.8.1. Code organization
The structure is different 
image

**Comment by emmurphy1** [comment]
> 10.1. Downloading and building the Spring Boot school timetable quick start
> 12.2.1. Preparing deployment files
> 14.2. Download and build the OptaWeb Vehicle Routing deployment files
> 10.1. Downloading and building the Spring Boot school timetable quick start
> This sections has common pre requirement - setup maven repo with following steps:
> 
> ```
> Procedure
> 
>     1. Navigate to the Software Downloads page in the Red Hat Customer Portal (login required), and select the product and version from the drop-down options:
>         Product: Red Hat Process Automation Manager
>         Version: 7.11 
>     2. Download Red Hat Process Automation Manager 7.11 Maven Repository (kogito-1.5.0-maven-repository.zip).
>     3. Extract the kogito-1.5.0-maven-repository.zip file.
>     4. Copy the contents of the kogito-1.5.0.GA-maven-repository/maven-repository subdirectory into the ~/.m2/repository directory. 
> ```
> 
> Could we move it in separate section and refer to it?

@dupliaka I see what you are saying, but in this case I don't think we should. It comes down to what is best for the user. We could make it a prerequisite and add a small section to the end of the getting started story but I think that section wouldn't fit very well there, and I think in almost all cases the user would have to go to that section first. Let me know if you disagree and we can discuss.

**Comment by emmurphy1** [comment]
> optaweb-vehicle-routing-distribution-7.52.0.Final-redhat-00004/sources

I made this change to all of the places that you listed. There are two other places that I did not change. Should I update those as well?
14.4. CONFIGURE AND RUN OPTAWEB VEHICLE ROUTING MANUALLY
14.7.3. Working with the OptaWeb Vehicle Routing front-end module

**Comment by emmurphy1** [comment]
> 14.8.1. Code organization
> The structure is different
> image

Jiri, Anna says the code layout is different for the Code Organization section for VRP. It looks like a flat structure. Do we even need this section now? If so, please help me update it.
http://file.ork.redhat.com/~emmurphy/BXMSDOC-7577-pam/#code_organization

**Comment by emmurphy1** [comment]
@dupliaka  I have updated the drafts with these comments except for two that I need Jiri's help with.

**Comment by dupliaka** [comment]
> optaweb-vehicle-routing-distribution-7.52.0.Final-redhat-00004/sources



> > optaweb-vehicle-routing-distribution-7.52.0.Final-redhat-00004/sources
> 
> I made this change to all of the places that you listed. There are two other places that I did not change. Should I update those as well?
> 14.4. CONFIGURE AND RUN OPTAWEB VEHICLE ROUTING MANUALLY
> 14.7.3. Working with the OptaWeb Vehicle Routing front-end module

Yes, please change it there as well

**Comment by dupliaka** [comment]
> 4.4. Dinner party should be removed

Thanks for changes
The side effects of this:
Some parts of migration guide reference to the dinner party code. However those are just examples of usage so we do not need to keep Dinner party for that.

Please mention that this example is removed in "Changes" chapter.

**Comment by dupliaka** [comment]
> > 14.8.1. Code organization
> > The structure is different
> > image
> 
> Jiri, Anna says the code layout is different for the Code Organization section for VRP. It looks like a flat structure. Do we even need this section now? If so, please help me update it.
> http://file.ork.redhat.com/~emmurphy/BXMSDOC-7577-pam/#code_organization

@emmurphy1 

Sorry, this comment was more not about the view of the structure but about content.
Websocket changed to rest and in service error added. That is all requested changes.
Could you please change it to
```
org.optaweb.vehiclerouting
├── domain
├── plugin          # Infrastructure layer
│   ├── persistence
│   ├── planner
│   ├── routing
│   └── rest
└── service         # Application layer
    ├── demo
    ├── distance
    ├── error
    ├── location
    ├── region
    ├── reload
    ├── route
    └── vehicle
```

File: doc-content/enterprise-only/vehicle-route-planning/backend-architecture-con.adoc [hunk 1]
```diff
@@ -23,10 +23,11 @@ org.optaweb.vehiclerouting
 │   ├── persistence
 │   ├── planner
 │   ├── routing
-│   └── websocket
+│   └── rest
```
Comment (yurloc) [inline doc-content/enterprise-only/vehicle-route-planning/backend-architecture-con.adoc:26]: @emmurphy1 Yup, this (and the `error` package) is the only necessary change. The flat structure from Anna's screenshot is just a different way her IDE displays the packages.

File: doc-content/enterprise-only/employee-rostering/er-deploy-script-proc.adoc [hunk 2]
```diff
@@ -5,13 +5,15 @@ You can deploy the Employee Rostering starter application to {OPENSHIFT} using t
...
+* OpenJDK {OPTAPLANNER_JDK_VERSION} or later is installed. Red Hat build of Open JDK is available from the https://access.redhat.com/jbossnetwork/restricted/listSoftware.html[Software Downloads] page in the Red Hat Customer Portal (login required).
+* Apache Maven {OPTAPLANNER_MAVEN_VERSION} or higher is installed. Maven is available from the https://maven.apache.org/[Apache Maven Project] website.
+
 * A `bash` shell environment is available on your local system.
 
 .Procedure
 . Download the `kogito-1.5.0-maven-repository.zip` file from the {PRODUCT_DOWNLOAD_LINK}[Software Downloads] page of the Red Hat Customer Portal.
 . Expand the downloaded archive.
-. Copy the contents of the `kogito-1.5.0.GA-maven-repository/maven-repository` subdirectory into the `~/.m2/repository` directory.
+. Navigate to the `optaweb-8.5.0.Final-redhat-00001/optaweb-employee-rostering` folder.
```
Comment (yurloc) [inline doc-content/enterprise-only/employee-rostering/er-deploy-script-proc.adoc]: @emmurphy1 is this intentional? I'm afraid the procedure stops making sense.

**Comment by yurloc** [comment]
> > optaweb-vehicle-routing-distribution-7.52.0.Final-redhat-00004/sources
> 
> > > optaweb-vehicle-routing-distribution-7.52.0.Final-redhat-00004/sources
> > 
> > 
> > I made this change to all of the places that you listed. There are two other places that I did not change. Should I update those as well?
> > 14.4. CONFIGURE AND RUN OPTAWEB VEHICLE ROUTING MANUALLY
> > 14.7.3. Working with the OptaWeb Vehicle Routing front-end module
> 
> Yes, please change it there as well

Nope. The latest optawebs upgraded to Quarkus and distributed in kogito-examples are 8.5.0.Final. Is 7.52.0.Final being distributed in RHDM 7.11 at all?

**Comment by dupliaka** [comment]
> > > optaweb-vehicle-routing-distribution-7.52.0.Final-redhat-00004/sources
> > 
> > 
> > > > optaweb-vehicle-routing-distribution-7.52.0.Final-redhat-00004/sources
> > > 
> > > 
> > > I made this change to all of the places that you listed. There are two other places that I did not change. Should I update those as well?
> > > 14.4. CONFIGURE AND RUN OPTAWEB VEHICLE ROUTING MANUALLY
> > > 14.7.3. Working with the OptaWeb Vehicle Routing front-end module
> > 
> > 
> > Yes, please change it there as well
> 
> Nope. The latest optawebs upgraded to Quarkus and distributed in kogito-examples are 8.5.0.Final. Is 7.52.0.Final being distributed in RHDM 7.11 at all?

@yurloc 
Are you sure?
We are talking here exactly about changing 
optaweb-vehicle-routing-distribution-7.52.0.Final-redhat-00004/sources to
kogito-1.5.0-decision-services-quickstarts/optaweb-8.5.0.Final-redhat-00001/optaweb-vehicle-routing

Because indeed RHDM and RHPAM are not going to distribute optawebs

**Review by dupliaka — APPROVED** [review]
LGTM

File: doc-content/enterprise-only/employee-rostering/er-deploy-script-proc.adoc (hunk 2 above)
Comment (emmurphy1) [reply doc-content/enterprise-only/employee-rostering/er-deploy-script-proc.adoc]: @yuloc I can't remember why I made that change, but I had a look and reorganized the section. Would you please confirm? Also, about the file names, would you please scan these drafts and see if I have the right or not:
http://file.ork.redhat.com/~emmurphy/BXMSDOC-7577-dm/
http://file.ork.redhat.com/~emmurphy/BXMSDOC-7577-pam/

**Comment by dupliaka** [comment]
Sorry I found that changes for https://issues.redhat.com/browse/BXMSDOC-7670 was not applied
I copy it here
12.2.2. Running the Employee Rostering starter application JAR file

in procedure

in step 1

change sources kogito-1.5.0-decision-services-quickstarts/optaweb-8.5.0.Final-redhat-00001/optaweb-employee-rostering

in step 2

path to

kogito-1.5.0-decision-services-quickstarts/optaweb-8.5.0.Final-redhat-00001/optaweb-employee-rostering/optaweb-employee-rostering-standalone/target

 

 12.3. Overview of the source code of the employee rostering starter application modules and key classes

need to add "optaweb-employee-rostering-standalone" module it contains assembly configurations for standalone application.

**Review by dupliaka — APPROVED** [review]
- [x] School time table Spring Boot
- [x]  Getting Started with OptaPlanner
- [x]  Final QE review: ER on OS
- [x]  Final QE review: Optaplanner migration from 7 to 8
- [x]  Overall QE Review: Employee Rostering on an IDE
- [x]  Overall QE Review: VRP
- [x]  QE Review: School timetable on Quarkus

File: doc-content/enterprise-only/employee-rostering/er-deploy-script-proc.adoc (hunk 2 above)
Comment (yurloc) [reply doc-content/enterprise-only/employee-rostering/er-deploy-script-proc.adoc]: @emmurphy1 looks good to me now.

**Pull request merged**